### PR TITLE
Object saving

### PR DESCRIPTION
**Pull request by Mogwai-TheFurry**

### CODE REVIEW

File: Source/ACE.Database/ShardDatabase.cs [hunk 1]
```diff
@@ -461,68 +569,252 @@ private bool SaveObjectInternal(DatabaseTransaction transaction, AceObject aceOb
...
+                        // update it
+                        transaction.AddPreparedUpdateStatement<ShardPreparedStatement, AceObjectPropertiesInt>(ShardPreparedStatement.UpdateAceObjectPropertyInt, prop);
+                    }
+                }
+                else if (prop.PropertyValue != null)
+                {
+                    transaction.AddPreparedInsertStatement<ShardPreparedStatement, AceObjectPropertiesInt>(ShardPreparedStatement.InsertAceObjectPropertiesInt, prop);
+                }
+            }
+
```
Comment (ddevec) [inline Source/ACE.Database/ShardDatabase.cs]: Its one thing to copy/paste a 1 line function (smelly but not too bad), I'm weary of all this copy pasta.

File: Source/ACE.Entity/AceObjectPropertiesAttribute.cs [hunk 2]
```diff
@@ -5,23 +5,57 @@
...
+        private uint _xpSpent = 0;
+        private ushort _ranks = 0;
+        private ushort _base = 0;
+        
+        [DbField("attributeId", (int)MySqlDbType.UInt16, IsCriteria = true, Update = false)]
         public ushort AttributeId { get; set; }
 
         [DbField("attributeBase", (int)MySqlDbType.UInt16)]
-        public ushort AttributeBase { get; set; }
+        public ushort AttributeBase
```
Comment (ddevec) [inline Source/ACE.Entity/AceObjectPropertiesAttribute.cs]: Its moments like this I miss C++ templates and C macros.  All these manual IsDirty copies...

File: Source/ACE/Network/GameEvent/Events/GameEventPlayerDescription.cs [hunk 3]
```diff
@@ -157,7 +160,7 @@ private void WriteEventBody()
                 }
             }
 
-            var propertiesDid = aceObj.DataIdProperties.Where(x => x.PropertyId < 9000).ToList();
+            var propertiesDid = aceObj.DataIdProperties.Where(x => x.PropertyId < 9000 && x.PropertyValue != null).ToList();
```
Comment (ddevec) [inline Source/ACE/Network/GameEvent/Events/GameEventPlayerDescription.cs]: Do we want to allow null properties into our AceObject and/or database?  This function seems to assume that they will propagate there.

Comment (Mogwai-TheFurry) [inline Source/ACE/Network/GameEvent/Events/GameEventPlayerDescription.cs]: The issue is that they can be null at runtime.  Null values will ultimately get deleted in the database, but that doesn't stop the game-owned objects from having them.

**Comment by Mogwai-TheFurry** [comment]
I was able to resolve the merge conflict in a web browser, but the problem is I changed some method declarations that Og used so it's just borked.  I'll try to rebase tonight.

**Comment by Mogwai-TheFurry** [comment]
Note: character creation is broken.  Do not merge.

**Pull request merged**